test/utilities.js

```diff
@@ -181,21 +182,24 @@ const UINT32_WIDTH = 4 * 2
 const UINT128_WIDTH = 16 * 2
 
 function decodeAuctionElements(bytes) {
+  if (!bytes) {
+    return []
+  }
   const result = []
   bytes = bytes.slice(2) // cutting of 0x
   while (bytes.length > 0) {
     const element = bytes.slice(0, 112 * 2).split("")
     bytes = bytes.slice(112 * 2)
     result.push({
       user: "0x" + element.splice(0, ADDRESS_WIDTH).join(""), // address is only 20 bytes
-      sellTokenBalance: parseInt(element.splice(0, UINT256_WIDTH).join(""), 16),
-      buyToken: parseInt(element.splice(0, UINT16_WIDTH).join(""), 16),
-      sellToken: parseInt(element.splice(0, UINT16_WIDTH).join(""), 16),
-      validFrom: parseInt(element.splice(0, UINT32_WIDTH).join(""), 16),
-      validUntil: parseInt(element.splice(0, UINT32_WIDTH).join(""), 16),
-      priceNumerator: parseInt(element.splice(0, UINT128_WIDTH).join(""), 16),
-      priceDenominator: parseInt(element.splice(0, UINT128_WIDTH).join(""), 16),
-      remainingAmount: parseInt(element.splice(0, UINT128_WIDTH).join(""), 16),
+      sellTokenBalance: new BN(element.splice(0, UINT256_WIDTH).join(""), 16),
+      buyToken: new BN(element.splice(0, UINT16_WIDTH).join(""), 16),
+      sellToken: new BN(element.splice(0, UINT16_WIDTH).join(""), 16),
+      validFrom: new BN(element.splice(0, UINT32_WIDTH).join(""), 16),
+      validUntil: new BN(element.splice(0, UINT32_WIDTH).join(""), 16),
+      priceNumerator: new BN(element.splice(0, UINT128_WIDTH).join(""), 16),
```

Comment:
`buyToken`, `sellToken`, `validFrom`, and `validUntil` should all be regular javascript `Number`s.

Reply:
Right... so I will do this asap, unless you wanna do it. I have got a few other things on the go at the moment. Could get to it by the end of the day

Reply:
done!